controlplane/src/core/services/Authentication.ts

```diff
@@ -28,6 +29,19 @@ export class Authentication implements Authenticator {
     private orgRepo: OrganizationRepository,
   ) {}
 
+  public async extractUserAndOrgId(headers: Headers): Promise<{ userId?: string; organizationId?: string }> {
+    try {
+      const user = await this.webAuth.authenticate(headers);
```

Comment:
Why do we need do this again? Everything should be available in the authcontext. Also some requests are issued by tokens, let's make this visible in the logs too. I recommend to pass the authcontext to `enrichLogger`.

Reply:
The logger enriching should happen outside the error handler for us to have information in the handler itself

Reply:
We need to find a solution without duplicating this. Based on your thoughts, we would also fail to get context when the `authenticate` fails for whatever reasons. I suggest a pragmatic approach e.g.

```ts
const logger = await enrichLogger(ctx, opts.logger, authContext);
```

If we ensure that this

```ts
const authContext = await opts.authenticator.authenticate(ctx.requestHeader);
```

is the first line of the handler, it has the same consistency as you proposed but without duplicating code.